code/modules/projectiles/updated_projectiles/guns/shotguns.dm

```diff
@@ -200,28 +200,27 @@ can cause issues with ammo types getting mixed up during the burst.
 						/obj/item/attachable/extended_barrel,
 						/obj/item/attachable/compensator,
 						/obj/item/attachable/magnetic_harness,
-						/obj/item/attachable/stock/tactical)
+						/obj/item/attachable/stock/tactical,
+						// /obj/item/attachable/verticalgrip,
+						// /obj/item/attachable/angledgrip,
+						// /obj/item/attachable/attached_gun/flamer,
+						// /obj/item/attachable/attached_gun/shotgun,
+						/obj/item/attachable/attached_gun/grenade,
+						)
 
 /obj/item/weapon/gun/shotgun/combat/New()
 	..()
-	attachable_offset = list("muzzle_x" = 33, "muzzle_y" = 19,"rail_x" = 10, "rail_y" = 21, "under_x" = 14, "under_y" = 16, "stock_x" = 14, "stock_y" = 16)
-	var/obj/item/attachable/attached_gun/grenade/G = new(src)
-	G.flags_attach_features &= ~ATTACH_REMOVABLE
-	G.attach_icon = "" //gun already has a better one
-	G.icon_state = ""
-	G.Attach(src)
-	update_attachable(G.slot)
-	G.icon_state = initial(G.icon_state)
+	attachable_offset = list("muzzle_x" = 33, "muzzle_y" = 19,"rail_x" = 10, "rail_y" = 21, "under_x" = 20, "under_y" = 14, "stock_x" = 14, "stock_y" = 16)
+	starting_attachment_types = list(/obj/item/attachable/attached_gun/grenade)
```

Comment:
This should go under `/obj/item/weapon/gun/shotgun/combat` instead of under the `New()` proc, like with the M41A.